tests/ProxyManagerTest/ProxyGenerator/Util/PropertiesTest.php

```diff
@@ -187,4 +187,45 @@ public function testGetInstanceProperties()
 
         $this->assertCount(9, $properties->getInstanceProperties());
     }
+
+    /**
+     * @dataProvider propertiesToSkipFixture
+     *
+     * @param string $propertyName with property name
+     * @param string $methodToGetProperties name of method to acces the property
+     */
+    public function testSkipPropertiesByFiltering($propertyName, $methodToGetProperties)
+    {
+        $properties = Properties::fromReflectionClass(
+            new ReflectionClass(ClassWithMixedProperties::class)
+        );
+
+        $this->assertArrayHasKey($propertyName, $properties->$methodToGetProperties());
+        $filteredProperties =  $properties->filter([$propertyName]);
+
+        $this->assertArrayNotHasKey($propertyName, $filteredProperties->$methodToGetProperties());
+    }
+
+    public function testPropertiesIsSkippedFromRelatedMethods()
+    {
+        $properties = Properties::fromReflectionClass(
+            new ReflectionClass(ClassWithMixedProperties::class)
+        );
+
+        $this->assertArrayHasKey("\0*\0protectedProperty0", $properties->getProtectedProperties());
+        $this->assertArrayHasKey("\0*\0protectedProperty0", $properties->getInstanceProperties());
+        $filteredProperties =  $properties->filter(["\0*\0protectedProperty0"]);
+
+        $this->assertArrayNotHasKey("\0*\0protectedProperty0", $filteredProperties->getProtectedProperties());
+        $this->assertArrayNotHasKey("\0*\0protectedProperty0", $filteredProperties->getInstanceProperties());;
+    }
+
+    public function propertiesToSkipFixture()
+    {
+        return [
+            ['publicProperty0', 'getPublicProperties'],
+            ["\0*\0protectedProperty0", 'getProtectedProperties'],
+            ["\0ProxyManagerTestAsset\\ClassWithMixedProperties\0privateProperty0", 'getPrivateProperties'],
```

Comment:
A test involving [`ClassWithCollidingPrivateInheritedProperties`](https://github.com/Ocramius/ProxyManager/blob/b3aecacb9b2fb11036aaea70ce44df9bff842052/tests/ProxyManagerTestAsset/ClassWithCollidingPrivateInheritedProperties.php) is required here in order to verify that filtering involves the class name as well.